[RFC] vim-patch.sh improvements  [ci skip]

blueyed opened the pull request:
IDEAS:

- [ ] include (git commit) summary line with `-l`, and `-L` (?).  The latter is meant for external parsing, so should be handled with care, but the former updates Vim sources always.  Could be done later still - not that trivial.

- [ ] have a mode to report missing patches for a given file, which might be based on Vim (with optional ".vim-src" prefix) or Neovim.  This should be distinct, since it is non-trivial to e.g. adjust the path in `$@` when passing it to git-log.

blueyed committed:
scripts/vim-patch.sh: fall back to "origin" for remote

Without this, it would fail e.g. with a locally cloned repo of Neovim.

justinmk commented on scripts/vim-patch.sh:
```diff
@@ -107,9 +107,12 @@ commit_message() {
-  git remote -v \
-    | awk '$2 ~ /github.com[:\/]neovim\/neovim/ && $3 == "(fetch)" {print $1; exit}'
+set_git_remote() {
+  git_remote=$(git remote -v \
```
this works on `local` variables?!

blueyed replied:
> this works on local variables?!

If the call is in the local scope, yes.

Will change it to echo "origin" then.

blueyed committed:
scripts/vim-patch.sh: assign_commit_details: handle tags  [ci skip]

- Handle "v" prefix from Vim tags.
- Exit in case of error therein already.

blueyed committed:
-l/-L: pass through git-log options  [ci skip]

This allows for only listing missing patches for a given Vim file:

> scripts/vim-patch.sh -L src/edit.c

TODO:

- [ ] remove/ignore ".vim-src/" prefix (for tab completion)?
- [ ] base on Neovim source files?  (output) - but not trivial/useful?!

blueyed commented:
Pushed an update to pass through git-log options with `-l` / `-L`.

blueyed commented:
I think updating sources should be done less often - at least e.g. with `-L`.
What do you think in this regard?
It could for example do it automatically with `-P` etc when the given Vim revision is not found only, but should do it with `-l` of course still.

While this takes some time in general, it is also more noisy and might even cause unexpected issues when you are editing/comparing a file from below `.vim-src`, and it gets updated due to use of `-L`, which was only meant to list patches really.

blueyed commented:
/cc @janlazo @erw7 
I think this is quite useful.  Any feedback?

janlazo commented:
@blueyed Can you add a sample line in the help message for this feature?

```sh
./scripts/vim-patch.sh -l -- src/testdir/runtest.vim
```
This filters the relevant patches for this file. Useful since I compare logs between Vim and Neovim on test files and I don't want to manually filter or keep a tab/split open for each log.

blueyed committed:
add example [ci skip]

janlazo commented:
It works with `-G` for filtering patches. Perhaps useful to note it in https://github.com/neovim/neovim/wiki/Merging-patches-from-upstream-Vim.
